messaging/WFLY-13793_disable_legacy_amq1_prefix.adoc

```diff
@@ -0,0 +1,128 @@
+
+=== QE Contacts
+
+tborgato@redhat.com
+
+=== Testing By
+[X] Engineering
+
+[ ] QE
+
+=== Affected Projects or Components
+* WildFly
+
+=== Other Interested Projects
+
+== Requirements
+
+=== Hard Requirements
+
+* Add a `enable-amq1-prefix`  boolean attribute to the https://wildscribe.github.io/WildFly/18.0/subsystem/messaging-activemq/external-jms-queue/[/subsystem=messaging-activemq/external-jms-queue resource] to control the enablement of Artemis 1.x prefix for queue defined on a remote Artemis server. Default is `true` (meaning that name retro-compatibility is enabled). For instance an queue named `testQueue`, by setting this attribute to `false`, it effectively allows you to create an external queue without amq1 prefix `jms.queue`. The created queue will be `testQueue`. Otherwise, it will create an external queue with amq1 prefix `jms.queue`. The created queue will be `jms.queue.testQueue`.
+
+
+* Add a `enable-amq1-prefix`  boolean attribute to the https://wildscribe.github.io/WildFly/18.0/subsystem/messaging-activemq/external-jms-topic/[/subsystem=messaging-activemq/external-jms-topic resource] to control the enablement of Artemis 1.x prefix for topic defined on a remote Artemis server. Default is `true` (meaning that name retro-compatibility is enabled). For instance a topic named `testTopic`, by setting this attribute to `false`, it effectively allows you to create an external topic without amq1 prefix `jms.topic`. The created topic will be `testTopic`. Otherwise, it will create an external topic with amq1 prefix `jms.topic`. The created topic will be `jms.topic.testTopic`.
+
+* For a destination (queue or topic) created by annotation `@JMSDestinationDefinition` on a remote Artemis Server, add a `enable-amq1-prefix` boolean property to control the enablement of Artemis 1.x prefix. If the property is not defined, the default is `true` (meaning that name retro-compatibility is enabled).
+
+[source, java]
+----
+@JMSDestinationDefinition(
+    // explicitly mention a resourceAdapter corresponding to a pooled-connection-factory resource to the remote server
+    resourceAdapter = "remote-artemis",
+    name="java:global/env/myQueue2",
+    interfaceName="javax.jms.Queue",
+    destinationName="myQueue2",
+    properties = {
+        "enable-amq1-prefix=false"
+    }
+)
+----
+
+=== Nice-to-Have Requirements
+
+N/A
+
+=== Non-Requirements
+
+N/A
+
+== Test Plan
+
+* WildFly integration testsuite `ExternalJMSDestinationManagementTestCase`
+
+Prepare Wildfly server messaging-activemq subsystem be able to use internal broker as remote one.
+Create queue `myExternalQueue` and `myExternalTopic` with `enable-amq1-prefix` set to `false`. Both created queues name should not contain amq1 prefix `jms.queue` or `jms.topic`.
+
+The opposite version of the test is: create queue `myExternalLegacyQueue` and `myExternalLegacyTopic` with `enable-amq1-prefix` set to `true`. Both created queue contains amq1 prefix `jms.queue` or `jms.topic`.
+
+
```

Comment:
@soul2zimate we also need a test to check what happens when both `messaging-activemq.pooled-connection-factory.enable-amq1-prefix` and `messaging-activemq.external-jms-queue.enable-amq1-prefix` are set and they have different vaues

Reply:
@soul2zimate this test would limit its scope to destination creation,why don't we add also sending some messages to the destination? this would make clear what is the value that we need on `messaging-activemq.pooled-connection-factory.enable-amq1-prefix` to complete the process successfully: I mean creating, sending ad receiving a message to the same queue making sure e.g. we don't create queue `myQueue` and the try to send a message to queue `jms.queue.MyQueue`

Reply:
In @ehsavoie 's tests, [ExternalJMSDestinationManagementTestCase](https://github.com/wildfly/wildfly/pull/14412/files#diff-252482f0c3167ecf201ab5fea058fea3442d25f28154a05b8829d8e885616f87R60) tests management operations and check the created destinations.

And the " sending/receiving some messages to the destination" is covered in [ExternalJMSDestinationDefinitionMessagingDeploymentTestCase](https://github.com/wildfly/wildfly/pull/14412/files#diff-c0f01ceb72a3c4f68c2adcff2e7d1f8db8d39f174238f06f93f9031db72bed0bR75) via a proper deployment with MDB and Servlet in order to create, send ad receive messages.